File: examples/example_qthread/worker.cpp
#include <math.h>
#include "worker.h"

void Worker::setStop(bool value)
{
    stop = value;
}

Worker::Worker(QObject *parent):QObject(parent)
{
    n = 0;
    result = 0;

}

void Worker::set_param(unsigned long n){
    this->n = n;
}


void Worker::process(){
    this->stop = false;
    this->result = 0;
    auto d = 1;
    if (n != 0)  d = n/100;
    else progress(100);

    for ( unsigned long i = 0; i<=n; i++){
        this->result += sin(i);
        if (i % d == 0)
            progress(i/d);

        if (stop)
            break;
    }

    finished();
}


double Worker::get_result() const{
    return this->result;
}


Worker::~Worker(){

}

File: examples/java_thread_gui/src/main/java/zabgu/java_thread_gui/Worker.java
package zabgu.java_thread_gui;

import static java.lang.Math.sin;

public class Worker {

    public int n = 1;
    private double result = 0.0;
    public volatile boolean stop = false;  // отвечает за остановку потока
    // volatile переменная может безопасно изменятся несколькими потоками


    Worker(int n1){
        if (n1 > 0)
            n = n1;
        else
            n = 1;
    }

    public void run(){
        stop = false;
        result = 0.0;
        for (int i =0; i < n; i++){
            if (!stop)
                result += sin(i);
            else
                return;
        }
    }

    public double getResult() {
        return result;
    }
}
